### PR TITLE
Prevent 1p requests from being CNAME uncloaked in standard shields mode

### PR DESCRIPTION
**Pull request by antonok-edm**
Followup to https://github.com/brave/brave-core/pull/9673
Resolves https://github.com/brave/brave-browser/issues/17366

#9673 worked in some cases, but doesn't cover the case where the 1p domain has a CNAME record. This is a small tweak to disable CNAME uncloaking when:
- The new default-1p-blocking feature is disabled
- The request was made from a context where "Trackers & Ads blocked" was set to "Aggressive" mode
- The request URL is first party to the initiator (using the same logic as the rest of the adblocking code)

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Copied from #9673

> Disable the new `#brave-adblock-default-1p-blocking` flag, then visit https://dev-pages.bravesoftware.com/filtering/network-requests.html and follow the directions,  ensuring that the images are shown as described.

### CODE REVIEW

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc [hunk 1]
```diff
@@ -291,6 +291,19 @@ void OnBeforeURLRequestAdBlockTP(const ResponseCallback& next_callback,
...
       ProxySettingsAllowUncloaking(ctx->browser_context);
 
+  // When default 1p blocking is disabled, first-party requests should not be
+  // CNAME uncloaked unless using aggressive blocking mode.
+  if (!base::FeatureList::IsEnabled(
+          brave_shields::features::kBraveAdblockDefault1pBlocking) &&
+      should_check_uncloaked && !ctx->aggressive_blocking &&
+      SameDomainOrHost(
+          ctx->request_url,
+          url::Origin::CreateFromNormalizedTuple("https",
```
Comment (iefremov) [inline browser/net/brave_ad_block_tp_network_delegate_helper.cc:301]: `ctx->initiator_url.GetOrigin()` ?

**Comment by iefremov** [comment]
we will need a test for this

**Review by iefremov — APPROVED** [review]

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc (hunk 1 above)
Comment (antonok-edm) [reply browser/net/brave_ad_block_tp_network_delegate_helper.cc:301]: I wanted to use the same logic here as we do in the rest of the adblock engines. We should investigate whether or not that change is appropriate in the other places too - although it'll probably happen with the larger restructuring ahead.

**Comment by antonok-edm** [comment]
Agreed on the tests, but sadly there's no way to produce a unittest for this at the moment because of `ctx->browser_context` not being mockable. Let's look into it as a followup.

Not sure if I mentioned, but I already added some tests for the original PR as well.

**Pull request merged**